src/Stratis.Bitcoin.Features.Wallet/WalletTransactionHandler.cs

```diff
@@ -416,11 +392,11 @@ public class TransactionBuildContext
         /// <param name="recipients">The target recipients to send coins to.</param>
         /// <param name="walletPassword">The password that protects the wallet in <see cref="accountReference"/></param>
         /// <param name="opReturnData">Optional transaction data <see cref="OpReturnData"/></param>
-        public TransactionBuildContext(WalletAccountReference accountReference, List<Recipient> recipients,
-            string walletPassword = "", string opReturnData = null)
+        public TransactionBuildContext(Network network, WalletAccountReference accountReference, List<Recipient> recipients, string walletPassword = "", string opReturnData = null)
         {
             Guard.NotNull(recipients, nameof(recipients));
 
+            this.TransactionBuilder = new TransactionBuilder(network);
```

Comment:
This feels like a smell. We pollute our code with `Network` but it's not an explicit dependency of `TransactionBuildContext`.

Reply:
It is an explicit dependency on `TransactionBuilder`, which must (currently) be set on `TransactionBuildContext`.

Reply:
Exactly, but given that TransactionBuilder needs to change so often it's likely not even a true class-level dependency.

I don't think this is a blocker to merging though.